internal/providers/terraform/aws/kinesisanalyticsv2_application_snapshot.go

```diff
@@ -0,0 +1,43 @@
+package aws
+
+import (
+	"strings"
+
+	"github.com/infracost/infracost/internal/schema"
+	"github.com/shopspring/decimal"
+	"github.com/tidwall/gjson"
+)
+
+func GetKinesisDataAnalyticsSnapshotRegistryItem() *schema.RegistryItem {
+	return &schema.RegistryItem{
+		Name:                "aws_kinesisanalyticsv2_application_snapshot",
+		RFunc:               NewKinesisDataAnalyticsSnapshot,
+		ReferenceAttributes: []string{"application_name"},
+	}
+}
+
+func NewKinesisDataAnalyticsSnapshot(d *schema.ResourceData, u *schema.UsageData) *schema.Resource {
+	region := d.Get("region").String()
+	costComponents := make([]*schema.CostComponent, 0)
+	var durableApplicationBackupGb *decimal.Decimal
+
+	applicationName := d.References("application_name")
+	runtimeEnvironment := applicationName[0].Get("runtime_environment").String()
+
+	if u != nil && u.Get("durable_application_backup_gb").Type != gjson.Null {
+		durableApplicationBackupGb = decimalPtr(decimal.NewFromInt(u.Get("durable_application_backup_gb").Int()))
+	}
+
+	if strings.HasPrefix(strings.ToLower(runtimeEnvironment), "flink") {
```

Comment:
Looks good overall.  I think we should remove this check, it looks like the existence of a snapshot means that it is flink and can be charged.

Reply:
Done